### PR TITLE
Fix nrfconnect compilation errors

### PR DESCRIPTION
**Pull request by vinitg1**
Problem
Nrfconnect build broken during backport to interop_testing_2022_01_01

Change overview
- IM server is now initialized w/o parameters
- Changes in KConfig
- Corrected the command to flash and erase

Testing
Tested on local gLinux build ./chef.py -bz -d lighting-app -t nrfconnect

### CODE REVIEW

File: examples/chef/nrfconnect/CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ include(${CHEF}/project_include.cmake)
 
 get_filename_component(GEN_DIR ${CHEF}/out/${SAMPLE_NAME}/zap-generated REALPATH)
 
-set(CONF_FILE prj.conf)
+set(CONF_FILE ${CHIP_ROOT}/config/nrfconnect/app/sample-defaults.conf prj.conf)
```
Comment (Damian-Nordic) [inline examples/chef/nrfconnect/CMakeLists.txt:28]: `sample-defaults.conf` not longer exists. Does it even compile?

File: examples/chef/nrfconnect/Kconfig [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #
 mainmenu "Matter nRF Connect Chef Example Application"
 
-rsource "../../../config/nrfconnect/chip-module/Kconfig.defaults"
-rsource "../../../config/nrfconnect/chip-module/Kconfig.features"
+#rsource "../../../config/nrfconnect/chip-module/Kconfig.defaults"
```
Comment (Damian-Nordic) [inline examples/chef/nrfconnect/Kconfig:18]: Why this change? you don't need to source `chip-module/Kconfig` explicitly as it will be loaded automatically by `Kconfig.zephyr`.

File: examples/chef/nrfconnect/prj.conf [hunk 3]
```diff
@@ -46,9 +46,6 @@ CONFIG_RESET_ON_FATAL_ERROR=n
 # Disable Matter OTA DFU
 CONFIG_CHIP_OTA_REQUESTOR=n
 
-# Disable QSPI NOR
```
Comment (Damian-Nordic) [inline examples/chef/nrfconnect/prj.conf:49]: Why is this change needed? QSPI NOR is only needed if you do the DFU.

**Review by Damian-Nordic — CHANGES_REQUESTED** [review]

File: examples/chef/nrfconnect/main.cpp [hunk 4]
```diff
@@ -87,9 +87,7 @@ CHIP_ERROR main()
     chip::app::DnssdServer::Instance().SetExtendedDiscoveryTimeoutSecs(kExtDiscoveryTimeoutSecs);
 
     // Start IM server
-    static chip::CommonCaseDeviceServerInitParams initParams;
-    (void) initParams.InitializeStaticResourcesBeforeServerInit();
-    ReturnErrorOnFailure(chip::Server::GetInstance().Init(initParams));
+    ReturnErrorOnFailure(chip::Server::GetInstance().Init());
```
[the remark this answers is not in the document]
Comment (vinitg1) [reply examples/chef/nrfconnect/main.cpp:90]: Please note that this is an interop branch and this changes were made because it wasn't building before.

Init on this interop branch doesn't have any parameters.

File: examples/chef/nrfconnect/prj.conf (hunk 3 above)
Comment (vinitg1) [reply examples/chef/nrfconnect/prj.conf:49]: This change was blocking the build.

File: examples/chef/nrfconnect/CMakeLists.txt (hunk 1 above)
Comment (vinitg1) [reply examples/chef/nrfconnect/CMakeLists.txt:28]: Please note this is for a special interop branch and not master. All these changes were done to make it compile on the branch.

Comment (Damian-Nordic) [reply examples/chef/nrfconnect/CMakeLists.txt:28]: Ok, sorry, I didn't notice that.

**Review by Damian-Nordic — APPROVED** [review]
Approving. I didn't notice that it's for the interop branch. The comment about `-c` flag still applies though.

**Comment by vinitg1** [comment]
> 

Thanks Damian. I have corrected it now.

**Review by andy31415 — APPROVED** [review]
